homeassistant/components/airthings_ble/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""The Airthings BLE integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from airthings_ble import AirthingsBluetoothDeviceData
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Airthings BLE device from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    address = entry.unique_id
+
+    elevation = hass.config.elevation
+    is_metric = hass.config.units.is_metric
+    assert address is not None
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, address)
+
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find Airthings device with address {address}"
+        )
+
+    async def _async_update_method():
+        """Get data from Airthings BLE."""
+        ble_device = bluetooth.async_ble_device_from_address(hass, address)
+        airthings = AirthingsBluetoothDeviceData(_LOGGER, elevation, is_metric)
+
+        try:
+            data = await airthings.update_device(ble_device)
+        except Exception as err:
```

Comment:
Normally we try to avoid catching broad exception outside of the config flow. Can we narrow this down to the exception we expect?